cpu/sam0_common/sam0_sdhc/mtd_sdhc.c

```diff
@@ -46,9 +46,11 @@ static int _init(mtd_dev_t *dev)
 
 #if IS_USED(MODULE_MTD_WRITE_PAGE)
     /* TODO: move to MTD layer */
```

Comment:
Is that TODO still valid?

Reply:
hm, `mtd_sam0_sdhc_driver` has the `MTD_DRIVER_FLAG_DIRECT_WRITE` flag set which means the MTD layer would not allocate the work area, but the driver needs it.

Reply:
The difference is that the `mtd_sam0_sdhc_driver.init` allocates a `work_area`, but `mtd_sdmmc_driver.init` does not.
Is this actually the work around for the original motivation of the PR?